armi/reactor/blocks.py

```diff
@@ -230,7 +230,11 @@ def getSmearDensity(self, cold=True):
         """
         fuels = self.getComponents(Flags.FUEL)
         if not fuels:
-            return 0.0  # Smear density is not computed for non-fuel blocks
+            # Smear density is not computed for non-fuel blocks
+            return 0.0
+        elif not self.getNumPins():
+            # Smear density is only defined for pinned blocks
+            return 0.0
```

Comment:
Tests? Maybe we already have tests for that touch these conditionals 🤷

Reply:
Totally fair.  Done.